osf/models/spam.py

```diff
@@ -266,3 +280,33 @@ def do_check_spam(self, author, author_email, content, request_headers, update=T
                 self.spam_data['who_flagged'] = 'oopspam'
                 self.spam_data['oopspam_data'] = oopspam_details
         return akismet_is_spam or oopspam_is_spam
+
+    def moderate_domains(self, content, confirm_spam=False):
+        domains = re.findall(
+            settings.DOMAIN_REGEX,
+            content
+        )
+        if not domains:
+            return False
+
+        from osf.models import NotableDomain
+        if confirm_spam and NotableDomain.has_spam_domain(content):
+            self.confirm_spam(save=True)
+        else:
+            self.add_domains_to_moderation_queue(domains)
+
+        return True
+
+    def add_domains_to_moderation_queue(self, domains):
+        from osf.models import NotableDomain, DomainReference
+        for domain in domains:
+            domain = urlparse(domain)
+            notable_domain, created = NotableDomain.objects.get_or_create(
```

Comment:
if we pass the `confirm_spam` flag through to this function, we can check the status of the domain here and optionally mark spam at the end of the loop in a single pass instead of calling the status function and having to do the potentially expensive `__in` query above.